### PR TITLE
Improvements for Housekeeping service

### PR DESCRIPTION
**Pull request by ViacheslavKlimov**
## Pull Request description

Interrupt task processing on timeout to avoid blocking processor thread

## General checklist

- [ ] You have reviewed the guidelines [document](https://docs.google.com/document/d/1wqcOafLx5hth8SAg4dqV_LV3un3m5WYR8RdTJ4MbbUM/edit?usp=sharing).
- [ ] [Labels](https://docs.github.com/en/issues/using-labels-and-milestones-to-track-work/managing-labels#about-labels) that classify your pull request have been added.
- [ ] The [milestone](https://docs.github.com/en/issues/using-labels-and-milestones-to-track-work/about-milestones) is specified and corresponds to fix version.  
- [ ] Description references specific [issue](https://github.com/thingsboard/thingsboard/issues).
- [ ] Description contains human-readable scope of changes.
- [ ] Description contains brief notes about what needs to be added to the documentation.
- [ ] No merge conflicts, commented blocks of code, code formatting issues.
- [ ] Changes are backward compatible or upgrade script is provided.
- [ ] Similar PR is opened for PE version to simplify merge. Crosslinks between PRs added. Required for internal contributors only.
  
## Front-End feature checklist

- [ ] Screenshots with affected component(s) are added. The best option is to provide 2 screens: before and after changes;
- [ ] If you change the widget or other API, ensure it is backward-compatible or upgrade script is present.
- [ ] Ensure new API is documented [here](https://github.com/thingsboard/thingsboard-ui-help)

## Back-End feature checklist

- [ ] Added corresponding unit and/or integration test(s). Provide written explanation in the PR description if you have failed to add tests.
- [ ] If new dependency was added: the dependency tree is checked for conflicts.
- [ ] If new service was added: the service is marked with corresponding @TbCoreComponent, @TbRuleEngineComponent, @TbTransportComponent, etc.
- [ ] If new REST API was added: the RestClient.java was updated, issue for [Python REST client](https://github.com/thingsboard/thingsboard-python-rest-client) is created.
- [ ] If new yml property was added: make sure a description is added (above or near the property).

### CODE REVIEW

File: application/src/main/java/org/thingsboard/server/service/housekeeper/HousekeeperService.java [hunk 1]
```diff
@@ -137,7 +138,8 @@ protected <T extends HousekeeperTask> void processTask(ToHousekeeperServiceMsg m
             if (e instanceof ExecutionException) {
                 error = e.getCause();
             } else if (e instanceof TimeoutException) {
-                error = new TimeoutException("Timeout after " + config.getTaskProcessingTimeout() + " seconds");
+                future.cancel(true); // interrupting the task
```
Comment (smatvienko-tb) [inline application/src/main/java/org/thingsboard/server/service/housekeeper/HousekeeperService.java:141]: how about finally section?

if (!future.isDone) future.cancel(true);

File: application/src/main/java/org/thingsboard/server/service/housekeeper/HousekeeperService.java [hunk 2]
```diff
@@ -137,7 +138,8 @@ protected <T extends HousekeeperTask> void processTask(ToHousekeeperServiceMsg m
...
+                error = new TimeoutException("Timeout after " + config.getTaskProcessingTimeout() + " ms");
```
Comment (smatvienko-tb) [inline application/src/main/java/org/thingsboard/server/service/housekeeper/HousekeeperService.java:142]: I suggest to add to the text a keyword for easy full test search or grep, like 
`Housekeeper timeout ...`
and maybe some basic detail - tenant id, task details

**Review by smatvienko-tb — COMMENTED** [review]
looks good to me. this fixes the issue
```
"housekeeper-task-processor-66-thread-1" - Thread t@935
   java.lang.Thread.State: WAITING
	at java.base@17.0.12/jdk.internal.misc.Unsafe.park(Native Method)
	- parking to wait for <530819f6> (a com.google.common.util.concurrent.CollectionFuture$ListFuture)
	at java.base@17.0.12/java.util.concurrent.locks.LockSupport.park(LockSupport.java:211)
	at com.google.common.util.concurrent.AbstractFuture.get(AbstractFuture.java:563)
	at com.google.common.util.concurrent.AbstractFuture$TrustedFuture.get(AbstractFuture.java:110)
	at org.thingsboard.server.service.housekeeper.processor.TsHistoryDeletionTaskProcessor.process(TsHistoryDeletionTaskProcessor.java:54)
	at org.thingsboard.server.service.housekeeper.processor.TsHistoryDeletionTaskProcessor.process(TsHistoryDeletionTaskProcessor.java:44)
	at org.thingsboard.server.service.housekeeper.HousekeeperService.lambda$processTask$1(HousekeeperService.java:138)
	at org.thingsboard.server.service.housekeeper.HousekeeperService$$Lambda$5623/0x00000070030ebdd0.call(Unknown Source)
	at java.base@17.0.12/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base@17.0.12/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base@17.0.12/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base@17.0.12/java.lang.Thread.run(Thread.java:840)

   Locked ownable synchronizers:
	- locked <7af59af9> (a java.util.concurrent.ThreadPoolExecutor$Worker)
```
image

image

File: application/src/main/java/org/thingsboard/server/service/housekeeper/HousekeeperService.java (hunk 2 above)
Comment (ViacheslavKlimov) [reply application/src/main/java/org/thingsboard/server/service/housekeeper/HousekeeperService.java:142]: Everything is logged already:
image

**Pull request merged**